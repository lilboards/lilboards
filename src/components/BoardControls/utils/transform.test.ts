import type { Column, Item } from 'src/types';

import {
  transformToCSV,
  transformToMarkdown,
  transformToRows,
} from './transform';

const column: Column = {
  createdAt: Date.now(),
  createdBy: 'user1',
  name: '',
} as const;

const item: Item = {
  createdAt: Date.now(),
  createdBy: 'user1',
  text: '',
} as const;

const transforms = [transformToCSV, transformToMarkdown, transformToRows];

describe.each(transforms)('%p', (transform) => {
  it('transforms empty columns', () => {
    expect(transform({}, {})).toMatchSnapshot();
  });

  it('transforms columns with no name', () => {
    expect(
      transform(
        {
          column1: column,
          column2: column,
        },
        {},
      ),
    ).toMatchSnapshot();
  });

  it('transforms column with no items', () => {
    expect(
      transform(
        {
          column1: {
            ...column,
            name: 'Column Name',
          },
        },
        {},
      ),
    ).toMatchSnapshot();
  });

  it('transforms columns with no items', () => {
    expect(
      transform(
        {
          column1: {
            ...column,
            name: 'Column Name 1',
          },
          column2: {
            ...column,
            itemIds: [],
            name: 'Column Name 2',
          },
        },
        {},
      ),
    ).toMatchSnapshot();
  });

  it('transforms column with item', () => {
    expect(
      transform(
        {
          column1: {
            ...column,
            itemIds: ['item1'],
            name: 'Column Name',
          },
        },
        {
          item1: {
            ...item,
            text: 'Item 1',
          },
        },
      ),
    ).toMatchSnapshot();
  });

  it('transforms column with items', () => {
    expect(
      transform(
        {
          column1: {
            createdAt: Date.now(),
            createdBy: 'user1',
            itemIds: ['item1', 'item2'],
            name: 'Column Name',
          },
        },
        {
          item1: {
            ...item,
            text: 'Item 1',
          },
          item2: {
            ...item,
            text: 'Item 2',
          },
        },
      ),
    ).toMatchSnapshot();
  });

  it('transforms columns with items', () => {
    expect(
      transform(
        {
          column1: {
            ...column,
            itemIds: ['item1'],
            name: 'Column 1',
          },
          column2: {
            ...column,
            itemIds: ['item2', 'item3'],
            name: 'Column 2',
          },
        },
        {
          item1: {
            ...item,
            text: 'Item 1',
          },
          item2: {
            ...item,
            text: 'Item 2',
          },
          item3: {
            ...item,
            text: 'Item 3',
          },
        },
      ),
    ).toMatchSnapshot();
  });

  it('transforms column with item with newlines', () => {
    expect(
      transform(
        {
          column1: {
            ...column,
            itemIds: ['item1'],
            name: 'Column Name',
          },
        },
        {
          item1: {
            ...item,
            text: 'Item\n\n1',
          },
        },
      ),
    ).toMatchSnapshot();
  });
});
