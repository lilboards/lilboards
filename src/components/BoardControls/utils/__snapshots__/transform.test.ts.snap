// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Function transformToCSV] transforms column with item 1`] = `
""Column Name"
"Item 1""
`;

exports[`[Function transformToCSV] transforms column with item with newlines 1`] = `
""Column Name"
"Item\\n\\n1""
`;

exports[`[Function transformToCSV] transforms column with items 1`] = `
""Column Name"
"Item 1"
"Item 2""
`;

exports[`[Function transformToCSV] transforms column with no items 1`] = `""Column Name""`;

exports[`[Function transformToCSV] transforms columns with items 1`] = `
""Column 1","Column 2"
"Item 1","Item 2"
,"Item 3""
`;

exports[`[Function transformToCSV] transforms columns with no items 1`] = `""Column Name 1","Column Name 2""`;

exports[`[Function transformToCSV] transforms columns with no name 1`] = `""Column 1","Column 2""`;

exports[`[Function transformToCSV] transforms empty columns 1`] = `""`;

exports[`[Function transformToMarkdown] transforms column with item 1`] = `
"| Column Name |
| --- |
| Item 1 |"
`;

exports[`[Function transformToMarkdown] transforms column with item with newlines 1`] = `
"| Column Name |
| --- |
| Item<br><br>1 |"
`;

exports[`[Function transformToMarkdown] transforms column with items 1`] = `
"| Column Name |
| --- |
| Item 1 |
| Item 2 |"
`;

exports[`[Function transformToMarkdown] transforms column with no items 1`] = `
"| Column Name |
| --- |"
`;

exports[`[Function transformToMarkdown] transforms columns with items 1`] = `
"| Column 1 | Column 2 |
| --- | --- |
| Item 1 | Item 2 |
|  | Item 3 |"
`;

exports[`[Function transformToMarkdown] transforms columns with no items 1`] = `
"| Column Name 1 | Column Name 2 |
| --- | --- |"
`;

exports[`[Function transformToMarkdown] transforms columns with no name 1`] = `
"| Column 1 | Column 2 |
| --- | --- |"
`;

exports[`[Function transformToMarkdown] transforms empty columns 1`] = `""`;

exports[`[Function transformToRows] transforms column with item 1`] = `
[
  [
    "Column Name",
  ],
  [
    "Item 1",
  ],
]
`;

exports[`[Function transformToRows] transforms column with item with newlines 1`] = `
[
  [
    "Column Name",
  ],
  [
    "Item

1",
  ],
]
`;

exports[`[Function transformToRows] transforms column with items 1`] = `
[
  [
    "Column Name",
  ],
  [
    "Item 1",
  ],
  [
    "Item 2",
  ],
]
`;

exports[`[Function transformToRows] transforms column with no items 1`] = `
[
  [
    "Column Name",
  ],
]
`;

exports[`[Function transformToRows] transforms columns with items 1`] = `
[
  [
    "Column 1",
    "Column 2",
  ],
  [
    "Item 1",
    "Item 2",
  ],
  [
    undefined,
    "Item 3",
  ],
]
`;

exports[`[Function transformToRows] transforms columns with no items 1`] = `
[
  [
    "Column Name 1",
    "Column Name 2",
  ],
]
`;

exports[`[Function transformToRows] transforms columns with no name 1`] = `
[
  [
    "Column 1",
    "Column 2",
  ],
]
`;

exports[`[Function transformToRows] transforms empty columns 1`] = `[]`;
